app/src/main/java/com/monkeyteam/chimpagne/ui/FindEventScreen.kt

```diff
@@ -0,0 +1,370 @@
+        Button(
+            onClick = { onSearchClick() },
+            modifier =
+                Modifier.fillMaxWidth().padding(8.dp).height(56.dp), // Typical height for buttons
+            shape = MaterialTheme.shapes.extraLarge) {
+              Icon(Icons.Rounded.Search, contentDescription = "Search")
+              Spacer(Modifier.width(8.dp))
+              Text("Search", style = MaterialTheme.typography.bodyLarge)
+            }
+      }) { innerPadding ->
+        Box(modifier = Modifier.padding(innerPadding)) {
+          Column(
+              modifier =
+                  Modifier.fillMaxSize().padding(16.dp).verticalScroll(rememberScrollState()),
+              horizontalAlignment = Alignment.Start) {
+                FindEventLegend("Event Location", Icons.Rounded.Search, "Search")
+
+                Spacer(Modifier.height(16.dp))
+
+                LocationFinder("", {}, emptyList(), {}, Modifier.fillMaxWidth())
+                Spacer(Modifier.height(16.dp))
+                IconTextButton(
+                    text = "Locate me",
+                    icon = Icons.Rounded.MyLocation,
+                    onClick = { /*TODO*/},
+                    modifier = Modifier.align(Alignment.CenterHorizontally))
+                Spacer(Modifier.height(16.dp))
+
+                Text(text = "Search Radius: ${searchRadius.toInt()} km")
+                Slider(
+                    value = searchRadius,
+                    onValueChange = { searchRadius = it },
+                    valueRange = 1f..30f,
+                    modifier = Modifier.fillMaxWidth())
+
+                Spacer(Modifier.height(40.dp))
+
+                FindEventLegend("Add Tags to your search", Icons.Rounded.Tag, "Tags")
+                Spacer(Modifier.height(16.dp))
+
+                TextField(
+                    value = tagInput,
+                    onValueChange = { tagInput = it },
+                    keyboardOptions = KeyboardOptions.Default.copy(imeAction = ImeAction.Done),
+                    keyboardActions =
+                        KeyboardActions(
+                            onDone = {
+                              if (tagInput.trim().isNotBlank()) {
+                                selectedTags.add(0, tagInput.trim().lowercase(Locale.getDefault()))
+                                tagInput = ""
+                              }
+                            }),
+                    maxLines = 1,
+                    placeholder = { Text("Enter a tag...") },
+                    modifier = Modifier.fillMaxWidth())
+
+                Spacer(Modifier.height(16.dp))
+
+                Row(modifier = Modifier.fillMaxWidth().horizontalScroll(rememberScrollState())) {
+                  selectedTags.forEach { tag -> TagChip(tag = tag) { selectedTags.remove(tag) } }
+                }
+
+                Spacer(Modifier.height(40.dp))
+
+                FindEventLegend("Date of the event", Icons.Rounded.CalendarToday, "Select date")
```

Comment:
Use strings defined in resources for text, this allows for translations

Reply:
Done